metadata-ingestion/src/datahub/ingestion/source/fs/s3_fs.py

```diff
@@ -0,0 +1,59 @@
+import boto3
+import smart_open
+from dataclasses import dataclass
+from datahub.ingestion.source.fs.fs_base import FileSystem, FileStatus
+from datahub.ingestion.source.fs.s3_list_iterator import S3ListIterator
+from urllib.parse import urlparse
+from typing import Iterable
+
+
+def parse_s3_path(path: str):
+    parsed = urlparse(path)
+    return S3Path(parsed.netloc, parsed.path.lstrip('/'))
+
+
+def assert_ok_status(s3_response):
+    is_ok = s3_response['ResponseMetadata']['HTTPStatusCode'] == 200
+    assert is_ok, f"Failed to fetch S3 object, error message: {s3_response['Error']['Message']}"
+
+
+@dataclass
+class S3Path:
+    bucket: str
+    key: str
+
+    def __str__(self):
+        return f"S3Path({self.bucket}, {self.key})"
+
+
+class S3FileSystem(FileSystem):
+
+    _s3 = boto3.client('s3')
+
+    @classmethod
+    def create_fs(cls):
+        return S3FileSystem()
+
+    def open(self, path: str, **kwargs):
+        transport_params = kwargs.update({'client': S3FileSystem._s3})
```

Comment:
how does an end user configure the s3 client?

Reply:
Based on [boto3 docs](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html) there are 3 options. But anyway I added ability to configure **S3FileSystem** using **Config object**